### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by marijanbrvar**
1. Add calculator.js and operate.js into a logic/ directory
2. Implemented event handlers to use the math logic from calculator.js and operate.js in the component.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello  @marijanbrvar 👋

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,46 +1,30 @@
 /* eslint-disable react/button-has-type */
+import PropTypes from 'prop-types';
 import { Component } from 'react';
+import BonusList from './ButtonList';
+import Display from './Display';
 
 // eslint-disable-next-line react/prefer-stateless-function
 export default class Calculator extends Component {
```
Comment (mikenath223) [inline src/components/calculator.js:8]: - [ ] There is an error when a number is divided by zero kindly fix it.

 |

File: src/components/calculator.js [hunk 2]
```diff
@@ -1,46 +1,30 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
   render() {
+    const { display, click } = this.props;
```
Comment (mikenath223) [inline src/components/calculator.js:15]: - [ ] Also if an operator is clicked on twice it throws an invalid number error. Kindly fix it.

 |

File: src/components/AutoScalingText.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    const availableWidth = parentNode.offsetWidth;
+    const actualWidth = node.offsetWidth;
+    const actualScale = availableWidth / actualWidth;
+
+    if (scale === actualScale) return;
+
+    if (actualScale < 1) {
+      this.scale().setState({ scale: actualScale });
+    } else if (scale < 1) {
+      this.scale().setState({ scale: 1 });
```
Comment (mikenath223) [inline src/components/AutoScalingText.js:28]: - This is neat. Nice idea creating an autoscale feature for the display text. 👍

**Review by bellom — APPROVED** [review]
## Hi @marijanbrvar  👏  

**Great work making all the necessary corrections from the previous review.**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**